aws-xray-recorder-sdk-core/src/main/java/com/amazonaws/xray/internal/RandomGenerator.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.xray.internal;
+
+import java.util.Random;
+import java.util.function.Supplier;
+
+/**
+ * An internal base class for unifying the potential randomness providers. In effect, it's a wrapper for
+ * a handful of {@code Random} methods.
+ *
+ * <p>This class is internal-only and its API may receive breaking changes at any time. Do not directly
+ * depend on or use this class.
+ */
+public abstract class RandomGenerator {
+    private final Supplier<Random> randomSupplier;
+
+    protected RandomGenerator(Supplier<Random> randomSupplier) {
+        this.randomSupplier = randomSupplier;
+    }
+
+    public final byte[] randomBytes(int bytes) {
```

Comment:
Since this is an internal class we can tie it to our requirements - let's have it return the base16 string itself. For secure random, `nextBytes` is much faster but for normal random, two `nextLong`s is faster, we should take advantage of that. If it's too much for this PR, we can address it later though no worries.